object/slicer/slicer.go

```diff
@@ -267,11 +250,26 @@ func initPayloadStream(ctx context.Context, ow ObjectWriter, header object.Objec
 		stubObject:       &stubObject,
 	}
 
-	maxObjSize := childPayloadSizeLimit(opts)
+	maxObjSize := opts.ObjectPayloadLimit()
+	if opts.payloadSizeKnown {
+		res.payloadSizeKnown = true
+		res.payloadSize = opts.payloadSize
+
+		if opts.payloadSize < maxObjSize {
+			maxObjSize = opts.payloadSize
+		}
+	}
+
+	if uint64(len(opts.payloadBuffer)) > maxObjSize {
+		res.buf = opts.payloadBuffer[:maxObjSize+1]
+	} else {
+		// TODO(#544): support custom memory allocators
+		res.buf = make([]byte, maxObjSize+1)
```

Comment:
so generally speaking no memory relief if not enough buffer is provided?

Reply:
gone